<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)" />

  <arg name="debug" default="false" />
  <arg name="standalone" default="true" />

  <arg name="target_frame" default="$(arg uav_name)/gps_origin" />

  <arg name="custom_config" default="" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="load" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="$(arg UAV_NAME)_nodelet_manager" />

  <group ns="$(arg uav_name)">

    <node name="cone_aggregator" pkg="nodelet" type="nodelet" args="standalone compton_cone_generator/ConeAggregator" output="screen" launch-prefix="$(arg launch_prefix)">

      <rosparam file="$(find compton_cone_generator)/config/cone_aggregator.yaml" command="load" />

      <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" />

      <param name="topic_name" value="compton_cone_generator/cones" />
      <param name="uav_name" value="$(arg uav_name)" />
      <param name="target_frame" value="$(arg target_frame)" />

      <!-- published topics -->
      <remap from="~cones_out" to="~cones" />

    </node>

  </group>

</launch>
